### ISSUE TITLE
Copper Join Huge Performance issue

### ISSUE THREAD

**Issue by alexis-bvq** [atoti/atoti#419]
## Steps to reproduce

Here is a notebook to reproduce.
When you had a copper join reference then measure is 40 times longer.

[BugCopperJoin.zip](https://github.com/atoti/atoti/files/7588996/BugCopperJoin.zip)

Too tricky to reproduce only in python code so I created a notebook.

## Environment

- atoti:
- Python:
- Operating System:

**Comment by Skyscimitar** [comment]
Hello,

After some investigation it turns out the performance issue isn't due to the many-to-many (copper) join, but to the addition of a `slicing` hierarchy to the cube. Having a slicing hierarchy changes the evaluation of the query because the default member of the slicing hierarchy is expressed in the location. Because `TimeStep` and `Quantile` are keys of the joined table, this forces the retrieval of the `TimeStep` field where `Quantile` is `0` (the default value). This changes the query plan and increases the number of intermediate results by the number of values of `TimeStep` (in your case 80). In your case, this doesn't change the result of the every `TimeStep` is present for each `Quantile`, however if there were only half of the time steps available for the quantile 0, then the result of your query would be different as well. Since atoti can't guess that all the values of `TimeStep` are present, it has no choice but to evaluate them, hence the increased execution time of the query.

**Comment by bloussou** [comment]
@alexis-bvq can we close this ?

**Comment by alexis-bvq** [comment]
yes we can, it is actually directly linked to this one : https://github.com/atoti/atoti/issues/381
I had to remove the feature from the demo because of the bad performance issues doing the workaround specified here.
